homeassistant/components/tado/__init__.py

```diff
@@ -52,6 +54,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     username = entry.data[CONF_USERNAME]
     password = entry.data[CONF_PASSWORD]
     fallback = entry.options.get(CONF_FALLBACK, CONST_OVERLAY_TADO_MODE)
+    if fallback not in CONST_OVERLAY_TADO_OPTIONS:
+        if fallback:
+            fallback = CONST_OVERLAY_TADO_MODE
+        else:
+            fallback = CONST_OVERLAY_MANUAL
```

Comment:
I think we should make sure that the config entry options has a default fallback that is in the `CONST_OVERLAY_TADO_OPTIONS`. Ie we should update the config entry if needed with `async_update_entry`. And also fix the default option in the options schema in the config flow.

Reply:
think we're good now?